### PR TITLE
feat: Extract conventional commit messages for the changelog

### PR DESCRIPTION
**Pull request by maelle**

### CODE REVIEW

File: tests/testthat/test-conventional-commits.R [hunk 1]
```diff
@@ -0,0 +1,7 @@
+test_that("Can parse conventional commits", {
+  repo <- withr::local_tempdir()
+  withr::local_dir(repo)
+  create_cc_repo(repo)
+  messages <- get_top_level_commits_impl(since = NULL)$message
+  expect_snapshot(collect_news(messages))
```
Comment (krlmlr) [inline tests/testthat/test-conventional-commits.R:6]: For pretty snapshots:

```suggestion
  expect_snapshot(writeLines(collect_news(messages)))
```

**Comment by maelle** [comment]
@krlmlr wouldn't it be worth really parsing the commit messages (maybe at a later point)

- translating the types into actual words
- translating the different BREAKING CHANGES markers (the breaking change trailers, the "!").
- grouping the h2 together (i.e. all features together for instance)
- ordering them according to a given order (probably features first, and within h2, breaking changes first)

The two latter items would be useful for a function related #147 to "merge" versions before a release.

File: R/conventional-commits.R [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# fix type commits should be translated to PATCH releases.
```
Comment (maelle) [inline R/conventional-commits.R:1]: that's a placeholder to hook this into `bump_version()` (some sort of automatic choice)

File: R/update-news.R [hunk 3]
```diff
@@ -14,13 +14,13 @@ collect_news <- function(messages) {
     cli_alert("Scraping {.field {length(messages)}} commit messages.")
   }
 
-  messages_lf <- gsub("\r\n", "\n", messages)
-  messages_nonempty <- messages_lf[messages_lf != ""]
-  messages_before_triple_dash <- map_chr(strsplit(messages_nonempty, "\n---", fixed = TRUE), 1)
-  message_lines <- strsplit(messages_before_triple_dash, "\n", fixed = TRUE)
-  message_bullets <- map(message_lines, keep, ~ grepl("^[*-]", .))
+  message_items <- messages %>%
```
Comment (maelle) [inline R/update-news.R:17]: using pipes to name less things

File: tests/testthat/_snaps/conventional-commits/NEWS.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Introduce a request id and a reference to latest request. Dismiss
+incoming responses other than from latest request.
+
+Remove timeouts which were used to mitigate the racing issue but are
+obsolete now.
+
+Reviewed-by: Z
+Refs: #123
+## feat(lang)
```
Comment (maelle) [inline tests/testthat/_snaps/conventional-commits/NEWS.md:17]: we might want empty lines before and after h2?

**Comment by maelle** [comment]
I need to look at the failures

**Comment by maelle** [comment]
@krlmlr now ready for review

File: tests/testthat/_snaps/conventional-commits/NEWS.md (hunk 4 above)
Comment (krlmlr) [reply tests/testthat/_snaps/conventional-commits/NEWS.md:17]: Yes, those would be great.

File: tests/testthat/_snaps/conventional-commits/NEWS.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+## docs
+ correct spelling of CHANGELOG
+
+## chore!
+ drop support for Node 6
+
+BREAKING CHANGE: use JavaScript features not available in Node 6.
+
+## feat(api)!
+feat(api)!: send an email to the customer when a product is shipped
```
Comment (krlmlr) [inline tests/testthat/_snaps/conventional-commits/NEWS.md:29]: I would have expected those to look a bit like the following, but let's tackle this in a future PR:

```suggestion
## docs

- correct spelling of CHANGELOG.

## chore

- drop support for Node 6.

## Breaking change

 - use JavaScript features not available in Node 6.

## Breaking change

### api

- send an email to the customer when a product is shipped

## feat

### api

- send an email to the customer when a product is shipped
```

**Review by krlmlr — APPROVED** [review]
Thanks. Overall, I think it would have been sufficient to do something like

```md
- docs: correct spelling

- chore!: drop support for Node 6

- BREAKING CHANGE: use JavaScript features not available in Node 6.

- ...
```

for the highlighted part. Baby steps and such. We can flesh out details later.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**